flow/core/kernel/scenario/base.py

```diff
@@ -275,22 +352,72 @@ def gen_random_start_pos(self, initial_config, num_vehicles, **kwargs):
             see flow/core/params.py
         num_vehicles : int
             number of vehicles to be placed on the network
-        kwargs : dict
-            extra components, usually defined during reset to overwrite initial
-            config parameters
 
         Returns
         -------
         list of tuple (float, float)
             list of start positions [(edge0, pos0), (edge1, pos1), ...]
         list of int
             list of start lanes
-        list of float
-            list of start speeds
         """
-        raise NotImplementedError
+        (x0, min_gap, bunching, lanes_distr, available_length,
+         available_edges, initial_config) = self._get_start_pos_util(
+            initial_config, num_vehicles)
+
+        # extra space a vehicle needs to cover from the start of an edge to be
+        # fully in the edge and not risk having a gap with a vehicle behind it
+        # that is smaller than min_gap
+        efs = min_gap + VEHICLE_LENGTH  # extra front space
+
+        for edge in available_edges:
+            available_length -= efs * min([self.num_lanes(edge), lanes_distr])
+
+        # choose random positions for each vehicle
+        init_absolute_pos = \
+            [random.random() * available_length
+             for _ in range(num_vehicles)]
+
+        # sort the positions of vehicles, for simplicity in using
+        init_absolute_pos.sort()
+
+        # these positions do not include the length of the vehicle, which need
+        # to be added
+        for i in range(num_vehicles):
+            init_absolute_pos[i] += (VEHICLE_LENGTH + min_gap) * i
+
+        decrement = 0
+        edge_indx = 0
+        startpositions = []
+        startlanes = []
+        for i in range(num_vehicles):
+            edge_i = available_edges[edge_indx]
+            pos_i = (init_absolute_pos[i] - decrement) % (
+                    self.edge_length(edge_i) - efs)
+            lane_i = int(((init_absolute_pos[i] - decrement) - pos_i) /
+                         (self.edge_length(edge_i) - efs))
 
-    def gen_custom_start_pos(self, initial_config, num_vehicles, **kwargs):
+            pos_i += efs
+
```

Comment:
This loop deserves a comment explaining what it does.

Reply:
yup, need to review this as well. Will create a new PR about this